src/components/toast.js

```diff
@@ -117,23 +117,36 @@ const Toast = forwardRef(({
   const toastRef = useRef(null)
   const [progress, setProgress] = useState(100)  
 
+  useEffect(() => {
+    if (open && timeOut) {
+      const t = setTimeout(() => {
+        if (onTimeOut) {
+          onTimeOut()
+        }
+        onClose()
+      }, timeOut)
+  
+      return () => clearTimeout(t)
+    }
+  }, [open, timeOut])
+
   useEffect(() => {
     if (timeOut > 0) {
       const toastEl = toastRef.current
       let fade = ''
 
-      if (open) {
-        interval = setInterval(() => {
-          setProgress((prev) => prev - 1)
-        }, (timeOut) / 100)
-
-        timer = setTimeout(() => {
-          if (onTimeOut) onTimeOut()
-          onClose()
-          clearInterval(interval)
-          setProgress(100)
-        }, timeOut + 500)
-      }
+      // if (open) {
+      // interval = setInterval(() => {
+      //   setProgress((prev) => prev - 1)
+      // }, (timeOut) / 100)
+
+      // timer = setTimeout(() => {
+      //   if (onTimeOut) onTimeOut()
+      //   onClose()
+      //   clearInterval(interval)
+      //   setProgress(100)
+      // }, timeOut + 500)
+      // }
```

Comment:
@kc-leung just commented out the relevant code blocks in `toast.js` to test its effects on the Toast component, if you approve of these suggestions, feel free to edit once merged